Library/Homebrew/linkage_checker.rb

```diff
@@ -3,12 +3,13 @@
 require "formula"
 
 class LinkageChecker
-  attr_reader :keg
+  attr_reader :keg, :formula
   attr_reader :brewed_dylibs, :system_dylibs, :broken_dylibs, :variable_dylibs
   attr_reader :undeclared_deps, :reverse_links
 
-  def initialize(keg)
+  def initialize(keg, formula=nil)
     @keg = keg
+    @formula = formula || resolve_formula
```

Comment:
I think I'd find it easier to understand if `resolve_formula` would accept a `keg` argument instead of operating on the `keg` attribute, but I don't feel strongly about this.